test/models/businesses/permission_test.rb

```diff
@@ -13,13 +13,20 @@ class Businesses::PermissionTest < ActiveSupport::TestCase
     refute permission.active_subscription?
   end
 
-  test "no active_subscription? if paused" do
+  test "no active_subscription? if paused and current_period_end don't have grace period" do
     customer = pay_customers(:one)
-    pay_subscriptions(:full_time).update!(pause_behavior: "void")
+    pay_subscriptions(:full_time).update!(pause_behavior: "void", pause_resumes_at: nil, current_period_end: 1.day.ago, pause_starts_at: 1.day.ago)
     permission = Businesses::Permission.new(customer.subscriptions)
     refute permission.active_subscription?
   end
 
+  test "active_subscription? if paused and current_period_end has grace period" do
+    customer = pay_customers(:one)
+    pay_subscriptions(:full_time).update!(pause_behavior: "void", pause_resumes_at: nil, current_period_end: 1.day.from_now, pause_starts_at: 1.day.from_now)
+    permission = Businesses::Permission.new(customer.subscriptions)
+    assert permission.active_subscription?
+  end
+
```

Comment:
I think we need one more test for the following:

```ruby
pay_subscriptions(:full_time).update!(pause_behavior: "void", pause_resumes_at: nil, current_period_end: 1.day.from_now, pause_starts_at: 1.week.ago)
permission = Businesses::Permission.new(customer.subscriptions)
assert permission.active_subscription?
```

If I understand correctly, this is an important test. It makes sure someone who paused (1 week ago) but is still on the grace period (expiring tomorrow) still has an active subscription.

Reply:
As far as I understand if current_period_end is 1 day from now so pause_stars_at will also be automatically set as same so it will also be 1 day from now and subscription access will be paused accordingly.

If we check here pause_start_date gets sets as per current_period_end date https://github.com/pay-rails/pay/blob/master/test/pay/stripe/subscription_test.rb#L219

so I think above mentioned test is conflicting if we set current_period_end from 1 day from now and pause_starts_date to one week before.

Also, when I paused ongoing subscription from stripe portal I observed that puase_starts_date in database was set same as that of current_period_end.

Let me know your thoughts if I am assuming it correctly.